invenio_cli/helpers/package_managers.py

```diff
@@ -171,3 +178,186 @@ def start_activated_subshell(self) -> List[str]:
+        ) -> ProcessResponse:
+            try:
+                module_name = module_pkg.package_json["name"]
+            except FileNotFoundError as e:
+                return ProcessResponse(
+                    error="No module found on the specified path. "
+                    f"File not found {e.filename}",
+                    status_code=1,
+                )
+
+            status_code = assets_pkg.link(module_name)
+            if status_code == 0:
+                return ProcessResponse(
+                    output="Global module linked correctly to local folder",
+                    status_code=0,
+                )
+            else:
+                return ProcessResponse(
+                    error=f"Unable to link module. Got error code {status_code}",
+                    status_code=status_code,
+                )
+
+        return [
+            AssetsFunction(_link_package_to_global, "Linking module to global dist..."),
+            AssetsFunction(_link_global_package, "Linking module to assets..."),
+        ]
+
+
+class PNPM(JavascriptPackageManager):
+    """Generate ``pnpm`` commands for managing JS packages."""
+
+    name = "pnpm"
+
+    def create_pynpm_package(self, package_json_path):
+        """Create a ``PNPMPackage`` with the path to ``package.json``."""
+        return PNPMPackage(package_json_path)
+
+    def install_local_package(self, path):
+        """Install the local JS package."""
+        return ["-C", str(path)]
+
+    def env_overrides(self):
+        """Provide environment overrides for building Invenio assets."""
+        return {"INVENIO_WEBPACKEXT_NPM_PKG_CLS": "pynpm:PNPMPackage"}
+
+    def package_linking_steps(self):
+        """Generate steps to link the target package to the project."""
+
+        def _prelink_dist(
+            assets_pkg: NPMPackage, module_pkg: NPMPackage
+        ) -> ProcessResponse:
+            """Execute the "prelink-dist" script."""
+            status_code = module_pkg.run_script("prelink-dist")
+            if status_code == 0:
+                return ProcessResponse(
+                    output="Successfully ran prelink-dist script.",
+                    status_code=0,
+                )
+            else:
+                return ProcessResponse(
+                    error=f"Unable to prelink-dist. Got error code {status_code}",
+                    status_code=status_code,
+                )
+
+        def _link_package_single_step(
+            assets_pkg: NPMPackage, module_pkg: NPMPackage
+        ) -> ProcessResponse:
+            """Execute the PNPM single-step package linking."""
+            try:
+                # Accessing `package_json` fails if the file can't be found
+                module_pkg.package_json["name"]
+
+                # This is geared towards Invenio JS packages...
+                # But so are all the other steps
+                status_code = assets_pkg.link(
+                    str(module_pkg.package_json_path.parent / "dist")
```

Comment:
the fact that we're linking against the `"dist"` directory and not using the scripts requires some knowledge about the target package... but so does executing the `link-dist` script, which is not exactly a standard AFAIK :sweat_smile: